src/arcsf/data/tofu.py

```diff
@@ -0,0 +1,193 @@
+            - if stratified == False represents fraction of total Qs
+        - random_seed: seed for reproducibility
+        - debug: returns dictionary containing meta_data if being used during testing
+    Returns:
+        - Two datasets with forget and retain sets
+            as well as a debugging dictionary (optional)
+    """
+
+    all_data = load_dataset("locuslab/TOFU", "full")["train"]
+
+    num_authors = 200  # hard coding author count for now
+    q_per_author = 20  # hard coding author question count for now
+
+    q_indices = [q for q in range(num_authors * q_per_author)]
+    a_indices = [math.floor(q / q_per_author) for q in q_indices]
+
+    all_data = all_data.add_column("question_index", q_indices)
+    all_data = all_data.add_column("author_index", a_indices)
+
+    # if author, then the number of questions to forget (per author) is all of them
+    if granularity == "author":
+        num_forget = q_per_author
+    # else: it depends on other arguments
+    elif granularity == "question":
+        if stratified:
+            # non-random stratification: hard code 4 questions for now
+            if not forget_random:
+                num_forget = int(4)
+            # random then it depends on q_per_author and forgotten_fact_fraction
+            elif forget_random:
+                num_forget = int(q_per_author * forgotten_fact_fraction)
+        # non stratified random: forgotten_fact_fraction applied across entire dataset
+        elif not stratified:
+            num_forget = int(num_authors * q_per_author * forgotten_fact_fraction)
+
+    debug_dict = {
+        "author_count": num_authors,
+        "q_per_author": q_per_author,
+        "num_forget": num_forget,
+    }
+
+    # Get full list of indices
+    all_indices = list(range(all_data.num_rows))
+
+    # Get full list of authors
+    authors = list(range(num_authors))
+
+    # Conditionally create full list of questions
+    if granularity == "author" or not forget_random:
+        forget_questions = list(range(num_forget))
+
+    # create lists of retained and forgotten authors in all cases (sometimes not used)
+    retain_authors, forget_authors = train_test_split(
+        authors,
+        test_size=forgotten_author_fraction,
+        random_state=random_seed,
+    )
+
+    debug_dict["forget_author_numbers"] = forget_authors
+
+    if granularity == "question":
+        if forget_random:
+            if stratified:
+                # if stratified and random: select train-test split within each
+                # set of author questions (currently selects same indices
+                # TODO: Is this the behaviour we want?
+                forget_indices = []
+                for author in forget_authors:
+                    all_author_indices = np.arange(
+                        get_forget_indices(author, 0, q_per_author),
+                        get_forget_indices(author, 20, q_per_author),
```

Comment:
Remove hardcoded 20?

```suggestion
                        get_forget_indices(author, q_per_author, q_per_author),
```